include/wx/private/window.h

```diff
@@ -33,6 +34,34 @@ inline wxSize GetAverageASCIILetterSize(const T& of_what)
     return s;
 }
 
+inline bool HasDPIChanged(const wxWindowBase* const win, wxSize& usedDPI)
+{
```

Comment:
I think we might need to do things differently depending on whether or not per-monitor DPI is supported elsewhere too, so I'd like to suggest adding some `extern bool wxHasPerWindowDPI()` that would be implemented in `src/msw/window.cpp` and writing this function as just
```cpp
inline bool HasDPIChanged(const wxWindowBase* const win, wxSize& usedDPI)
{
    if ( win && wxHasPerWindowDPI() )
    {
        const wxSize dpi = win->GetDPI();
        if ( dpi != usedDPI )
        {
            usedDPI = dpi;
            return true;
        }
    }

    return false;
}
```

Reply:
A disadvantage of your example is that `wxHasPerWindowDPI` has to be implemented for each port. I can't use a base function and override it just for MSW. And adding another (private) function to `wxWindow` or `wxTopLevel` seems a bit unnecessary for just a one-time use.

Maybe add `wxHasPerWindowDPI` as a separate function in `wx/private/window.h`?

Reply:
Isn't this code MSW-specific anyhow? It's in `wx/msw/private.h` after all -- or can it be used by wxQt too here?

Reply:
Its in the common `wx/private/window.h` because it has to be used from `common/wincmn.cpp` and `common/sizer.cpp`.
I'd rather add one `__WXMSW__` guard in this header than multiple in the cpp files.

Reply:
Sorry, I'm blind, I've somehow seen `msw` in the file name which is not there :-(

In any case, I'm fine with having `#ifdef __WXMSW__` here, I just thought it was unnecessary for some unfathomable reason.

I'd still use an early return instead of an extra bool variable, but this is just a minor style preference, of course (the code seems more clear like this to me).